First version.

Bjjyx opened the pull request:
Первая вресия. 
В целом работает, но не покидает мысль, что можно улучшить, но пока никак не могу придумать как.

Bjjyx committed:
First version.

doc-ODA reviewed (commented):
Привет! В целом не плохо, но есть несколько недочетов и некорректная работа при вводе определенных символов. Подскажу на что обратить внимание.

doc-ODA commented on src/main/kotlin/screens/ChooseScreen.kt:
```diff
@@ -0,0 +1,48 @@
+
+
+abstract class ChooseScreen<T: Storage>(open val array: ArrayList<T>){
+    var scanner = Scanner(System.`in`)
```
инициализацию сканера можно вынести в main, и передавать его в нужные экраны.

Bjjyx replied:
В целом идея была в том, чтобы сделать класс ChooseScreen максимально самостоятельным, поэтому сканер инициализировался в нем.

doc-ODA commented on src/main/kotlin/screens/ChooseScreen.kt:
```diff
@@ -0,0 +1,48 @@
+
+    abstract fun choose(numbOfArray: Int)
+
+    fun userInterface(){
```
по поводу названий функций, желательно что бы это были глаголы(или фраза содержащая глагол) и по названию должно быть понятно, что делает эта функция
https://kotlinlang.ru/docs/coding-conventions.html

doc-ODA commented on src/main/kotlin/screens/FirstScreen.kt:
```diff
@@ -0,0 +1,30 @@
+    override fun add() {
+        println("Введите название архива")
+        commandStr = scanner.nextLine()
+        if(commandStr.isEmpty()){
```
А если ввести "   " -одни пробелы, название будет пустым) то же и с остальными isEmpty
isBlank я думаю лучше вариант или trim() к commandStr 
Если isBlank не знаком,  почитай сравнение с isEmpty

doc-ODA commented on src/main/kotlin/screens/ChooseScreen.kt:
```diff
@@ -0,0 +1,48 @@
+        while (commandInt != endOfArray()){
+            printList()
+            try {
+                commandInt = scanner.nextInt()
```
а вот тут уже приложение работает не корректно, попробуй ввести " 0 0 "
Пройдись дебагом, попробуй самостоятельно понять, почему ошибка. 
И напиши, почему ты используешь scanner.nextInt(), а не nextLine()
Корректно ли работает приложение со scanner.skip

Bjjyx replied:
Если я правильно понял, при вводе " 0 0 " scanner.nextInt() пропускает пробел, записывает в память 0, потом считывает \n как "ввод" и вновь пропуская пробел записывает 0, но в моём случае мы использвуем пропуск \n, из-за чего программа и "вешается", если я правильно всё понял.

doc-ODA replied:
Ну почти, nextInt() действительно считает 0 и вызовется scanner.skip("\n"), тут программа и как ты выразился "вешается") Суть в том, что skip ожидает перенос строки, а там пробел. В общем он его так и не дождется, но если бы было .*\n оно бы отработало корректно, но ты молодец исправил.

Bjjyx committed:
Second version.

Исправлен баг с " 0 0 ". Тепер Scanner.nextInt() не использвуется, всё работает через nextLine() и преобразование строк в целое число, где это необходимо.
Исправлена возможность вписать название или текст в Note состоящие из одних пробелов.
Изменены названия большинства классов и функций
+ Небольшие доработки

doc-ODA reviewed (commented):
Все отлично, ошибки поправил!
